pkg/types/baremetal/validation/platform.go

```diff
@@ -32,6 +34,30 @@ func validateIPNotinMachineCIDR(ip string, n *types.Networking) error {
 	return nil
 }
 
+func validateOSImageURI(uri string, osimage string) error {
+	// Check for valid URI and sha256 checksum part of the URL
+	if strings.HasPrefix(uri, "http://") || strings.HasPrefix(uri, "https://") {
+		parsedURL, err := url.ParseRequestURI(uri)
+		if err != nil {
+			return fmt.Errorf("The %s URI provided is invalid", osimage)
```

Comment:
golint check is failing due to this and errors below, they need to begin with lowercase letters

Reply:
Thanks for catching this. I've fixed it in the latest (squashed) commit.